mpi4jax/_src/jax_compat.py

```diff
@@ -59,8 +76,34 @@ def check_jax_version():
     from jax.interpreters.mlir import token_type  # noqa: F401
 
 
+# TODO: remove this code once we only support jax > 0.4.14
+if versiontuple(jax.__version__) >= (0, 4, 14):
+    from jax.core import ShapedArray  # noqa: F401
+else:
+    from jax.abstract_arrays import ShapedArray  # noqa: F401
+
+
+# TODO: remove this code once we only support jax > 0.4.14
```

Comment:
```suggestion
# TODO: remove this code once we only support jax > 0.4.16
```

Reply:
I really meant `> 0.4.14` : `0.4.15` has been yanked so it is starting with 0.4.16... I'll just change this to `>0 0.4.16` which is more clear.